domain/secret/state/state.go

```diff
@@ -1494,8 +1519,10 @@ func (st State) listSecretRevisions(ctx context.Context, tx *sqlair.TX, uri *cor
 	query := `
 SELECT (sr.*) AS (&secretRevision.*),
        (svr.*) AS (&secretValueRef.*),
-       (sre.*) AS (&secretRevisionExpire.*)
+       (sre.*) AS (&secretRevisionExpire.*),
+       (sro.*) AS (&secretRevisionObsolete.*)
 FROM   secret_revision sr
+INNER JOIN secret_revision_obsolete sro ON sro.revision_uuid = sr.uuid
```

Comment:
This can be a left join and "obsolete" and "pending_delete" will default to false.